src/components/Dashboard.vue

```diff
@@ -683,65 +567,26 @@ export default Vue.extend({
       this.isOnboarding = false;
     },
     parseUserData(data: FirestoreNestedUserData, name: FirestoreUserName): AppUser {
-      const user: AppUser = {
-        // TODO: take into account multiple majors and colleges
-        college: data.colleges[0].acronym,
-        collegeFN: data.colleges[0].fullName,
-        firstName: name.firstName,
-        middleName: name.middleName,
-        lastName: name.lastName,
-        major: [],
-        majorFN: [],
-        minor: [],
-        minorFN: [],
-        exam: [],
-        transferCourse: [],
-        tookSwim: data.tookSwim,
-      };
-      const transferClasses = [];
-      if ('exam' in data && data.exam.length > 0) {
-        const exams: FirestoreAPIBExam[] = [];
-        data.exam.forEach(exam => {
-          // TODO: add a course to chosen requirement or multiple fulfilling requirements
-          exams.push(exam);
-          if ('equivCourse' in exam) {
-            transferClasses.push(exam.equivCourse[0]);
-          }
-        });
-        user.exam = exams;
-      }
-      if ('class' in data && data.class.length > 0) {
-        const classes = [];
-        for (const course of data.class) {
-          classes.push(course);
-          const courseInfo = this.createCourse(course.course);
-          transferClasses.push(courseInfo);
-          // ; // TODO for user to pick which req a class goes for
+      const user = createAppUser(data, name);
+
+      const transferClasses: any[] = [];
+      user.exam.forEach(exam => {
+        if ('equivCourse' in exam) {
+          transferClasses.push(exam.equivCourse[0]);
         }
-        user.transferCourse = classes;
-        this.currentClasses = transferClasses;
-      }
+      });
+      user.transferCourse.forEach(course => {
+        const courseInfo = firestoreCourseToAppCourse(
+          course.course,
+          () => this.incrementID(),
+          (subject) => this.addColor(subject)
+        );
+        transferClasses.push(courseInfo);
+        // ; // TODO for user to pick which req a class goes for
+      });
+      this.currentClasses = transferClasses;
```

Comment:
I decide to leave transfer course processing here for now. I'm not sure how it is used at this point, but just kept to preserve old behavior.